iceoryx_posh/source/popo/building_blocks/typed_unique_id.cpp

```diff
@@ -29,8 +30,7 @@ void setUniqueRouDiId(const uint16_t id) noexcept
 {
     if (hasDefinedUniqueRouDiId)
     {
-        errorHandler(
-            Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_UNIQUE_ID, [] {}, ErrorLevel::MODERATE);
+        errorHandler(Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_UNIQUE_ID, nullptr, ErrorLevel::MODERATE);
```

Comment:
This is quite error prone. Shall I create an issue to swap the order of the error callback and error level parameter?

This would then become
```cpp
errorHandler(Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_UNIQUE_ID, ErrorLevel::MODERATE);
```
I guess it is more common to just use a non fatal error level than to use a custom error callback.

Another option would be the builder pattern
```cpp
errorHandler(Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_UNIQUE_ID)
    .withErrorLevel(ErrorLevel::MODERATE)
    .customErrorCallback([]{
        std::cout << "You are doomed" << std::endl;
        std::terminate();
    });
```

What do you think?

Reply:
I prefer the builder pattern. It is the most expressive approach.

Reply:
I did not create an issue for this since @mossmaurice tackles this as part of #590 ... oh, I guess the name of that issue should be changed :)